### PR TITLE
Add regression test for TG-2479

### PR DESCRIPTION
**Pull request by yumibagge**
Add a test that failed initially and fixed by `smowton/cbmc/smowton/feature/synthesise-lambda-classes`

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 0db7ffa76447b3c8af4b55c770ef3bd4b1d3dd19).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/122813182
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

File: jbmc/regression/jbmc/lambda4/LambdaBasic.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      }
+      assert endOnSuccess; // verification failure
+      return 0;
+    };
+
+    int x0 = boo.xInt();
+    if (x0 == 1) {
+      assert x0 == 0; // verification failure
+    } else {
+      assert x0 >= -1 || x0 < -1; // verification success
```
Comment (smowton) [inline jbmc/regression/jbmc/lambda4/LambdaBasic.java]: `assert true`? Assert something nontrivial that you expect to pass instead.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 8a56cabe706a2253ff4db0288fc3630334d1e5d3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/122860324
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Comment by yumibagge** [comment]
@LAJW @smowton, the assertions are pretty much from Lukasz's but hopefully this is now satisfactory.  Please review.

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4b296f2d4b78a9c9a83f1870a7bdcd727cde6b1e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124743538

**Review by LAJW — APPROVED** [review]

**Pull request merged**